pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -363,6 +372,31 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 	return result, nil
 }
 
+// Record reboot requests
+func checkUpdatedRebootRequest(host *metal3v1alpha1.BareMetalHost) (dirty bool) {
+	requestTime := host.Spec.RequestRebootAt
+	if requestTime.IsZero() ||
+		requestTime.Equal(host.Status.PendingRebootSince) {
+		return
+	}
+
+	// Bring forward future-dated requests to the present
```

Comment:
If something schedules a reboot for the future, shouldn't we wait to do it?

Reply:
Maybe. It's not really an intended use of the API, so it seemed safer not to allow it.
The problem is that there's no way to cancel a request (although you can always request it to happen sooner), so if you request a reboot next year then next year the server will reboot and if you change your mind at any point in that year there's nothing you can do about it.
I'm happy to discuss though, because I don't know what the Right Thing would be.

Reply:
To cancel a future request wouldn't I just need to clear that timestamp field by setting it back to nil?

Reply:
The code that copies the `Spec.RequestRebootAt` field to `Status.PendingRebootSince` only does so if it's an earlier but non-zero time. Otherwise I think someone just updating the CRD from a yaml file could accidentally cancel any pending reboots.

Reply:
Isn't being able to cancel a reboot that is scheduled for the future ("9 am tomorrow") a feature?

Reply:
Indeed, it's a feature you'd certainly want if you had the feature of scheduling a future reboot - along with several other features, like the ability schedule multiple reboots at different times, and to configure whether a reboot prior to the scheduled time counts or not. The only way to implement those features (without requiring client-side merging logic) would be to create a separate BareMetalHostScheduledRebootRequest CRD that waits until the scheduled time and then triggers this interface. The more I think about it, the more that seems like the right way to implement a scheduled reboot if somebody actually requests it (and until then YAGNI).

Reply:
Sure, that's just like the remediation controller proposal.

If we're not actually going to manage the schedule, I still think it makes sense to just wait for the requested time instead of doing this bring-forward logic. That will make it behave the way the field names imply.

Reply:
Maybe the name is the problem? We can't control what users put in there, but my intention was to telegraph the fact that they should be putting the current time (in the case of fencing it would be the time of the fencing decision, for example). Maybe `rebootRequestTimestamp` would be better? I picture it working similarly to the `deletedTimestamp`, though of course you generally don't interact with that one directly thanks to some syntactic sugar in the client.